### ISSUE TITLE
Counterintuitive behavior with the bar cursor

### ISSUE THREAD

**Issue by dstrelnikov** [helix-editor/helix#1977]
### Summary

I prefer the bar cursor shape, however the selection looks counterintuitive. I'm not sure this will be considered as a bug cause the cursor is supposed to select a single character in helix.
![Screenshot_20220406_114258](https://user-images.githubusercontent.com/14166481/161948384-ed313c48-77a4-48bd-aea9-b7d12eb2e183.png)
 

### Reproduction Steps

Set
```
[editor.cursor-shape]
normal = "bar"
```

Open helix and type a word. Then switch back to the normal mode. Press `b` and see the first cursor-selection issue. Press `e` and see a similar issue.

### Platform

Linux

### Terminal Emulator

Konsole

### Helix Version

2.05-dev (deb7ee65)

**Comment by archseer** [comment]
This is because the selection shifts by one space when insert mode is entered. Would need to adjust rendering in normal mode.

**Comment by pascalkuthe** [comment]
This was partially fixed by #5575. The only remaining issue is that we might want to move the bar cursor to the right by one when it's the head similar to insert mode so it appears at the end of the selection instead of at the start of the last selected cursor.

**Comment by varaskkar** [comment]
> The only remaining issue is that we might want to move the bar cursor to the right by one when it's the head similar to insert mode

Yes please!! This fix would be great. I use insert mode a lot and this bug is really pesky.

I have this keymap in my config:

`m = ["trim_selections", "search_selection", "search_next"]`

When I wanna search for the char `;` I press the "m" key when the cursor is to the left of the ";" and it works. But, if I select the char `;` then it actually selects `;s` with `s` being the next character, which it shouldn't pick up.

![gif](https://i.imgur.com/2lUScRf.gif)
